contracts/cw20-base/src/contract.rs

```diff
@@ -14,19 +17,34 @@ use crate::allowances::{
 use crate::enumerable::{query_all_accounts, query_all_allowances};
 use crate::error::ContractError;
 use crate::msg::{ExecuteMsg, InstantiateMsg, QueryMsg};
-use crate::state::{MinterData, TokenInfo, BALANCES, TOKEN_INFO};
+use crate::state::{MarketingInfo, MinterData, TokenInfo, BALANCES, LOGO, TOKEN_INFO};
 
 // version info for migration info
 const CONTRACT_NAME: &str = "crates.io:cw20-base";
 const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
 
+const LOGO_SIZE_CAP: usize = 5 * 1024;
+
+fn verify_logo(logo: &Logo) -> Result<(), ContractError> {
+    if matches!(
+        logo,
+              Logo::Embedded(EmbeddedLogo::Svg(logo))
```

Comment:
It would good to also verify the format. At least somewhat...

eg. svg starts with `<?xml` and ends with `</svg>` (or a slightly less restrictive contains search on this). Maybe some other items

png should at least have the magic number (4 bytes prefix). 

We don't need to protect against people seeing malformed data, but rather check if they tried to shove a gif as a png. Better to error on upload.

Reply:
I was considering it, but abandoned as I decided that I am not able to perform validation in reasonable time (in terms of performance, not development time), and don't see point of such non-whole checks. In particular for svg to have reasonable check we should parse xml, as at the and `</svg>` doesn't need to be at the end (even comments may follow it), on the other hand - searching for `</svg>` may end up finding one which is specifically in the comment. Basically if someone would like to use malformed image as attack vector on someone using logo API, then we cannot protect it. The most we can actually do in reasonable time is checking xml preamble (as it shouldn't contain anything before it including comments), and probably header for png. I can add it.